Modules/mathmodule.c

```diff
@@ -1229,32 +1233,36 @@ FUNC1AD(gamma, m_tgamma,
       "gamma($module, x, /)\n--\n\n"
       "Gamma function at x.",
       "expected a float or nonnegative integer, got %s")
```

Comment:
I get error for `0`, which is a non-negative integer:
```pycon
>>> math.gamma(0)
Traceback (most recent call last):
  File "<python-input-1>", line 1, in <module>
    math.gamma(0)
    ~~~~~~~~~~^^^
ValueError: expected a float or nonnegative integer, got 0.0
```
I get also error for negative float value which happens to be an integer. The error message is not clear about this.
```pycon
>>> math.gamma(-1.0)
Traceback (most recent call last):
  File "<python-input-2>", line 1, in <module>
    math.gamma(-1.0)
    ~~~~~~~~~~^^^^^^
ValueError: expected a float or nonnegative integer, got -1.0
```

Reply:
> math.gamma(0)

Thanks, indeed - this should be positive, as for lgamma().

> I get also error for negative float value which happens to be an integer. The error message is not clear about this.

I don't see a problem here: the message says we expect nonnegative (>=0) input here (actually, it should be just positive) and you pass a negative integer float value.

Reply:
-1.0 is an integer in sense of `is_integer()` predicate or ``==``.

Reply:
But is is also a float. "a or b" is true if "a" is true.

Reply:
"expected a noninteger or positive integer, got %s"?

Reply:
Something like this, or via the complement of the function domain. It raises ValueError only for non-positive integers, isn't?

OverflowError with generic error message is raised for other values.